examples/consumer/main.go

```diff
@@ -1,21 +1,50 @@
 package main
 
 import (
-	"net/http"
+	"log"
 	_ "net/http/pprof"
+	"path/filepath"
+	"runtime"
+	"sync"
 	"time"
 
 	"github.com/retail-ai-inc/beanq"
+	"github.com/spf13/viper"
 )
 
-func main() {
+var (
+	configOnce sync.Once
+	bqConfig   beanq.BeanqConfig
+)
+
+func initCnf() beanq.BeanqConfig {
+	configOnce.Do(func() {
+		var envPath string = "./"
+		if _, file, _, ok := runtime.Caller(0); ok {
```

Comment:
`runtime.Caller(0)` returns information about the call stack, not the working directory. The file name and line number of course refer to the source code used to compile the executable. The compiler placed that info into the executable while creating the executable from source code. This information is available at runtime because it is contained (as metadata) in the executable binary.

So do you think it is safe to use `runtime.Caller(0)` because the binary you may compile some other machine but run it in GCP?